src/requests/parameter_wrappers.rs

```diff
@@ -1,235 +1,188 @@
+            &get_element!(Array),
+        )?),
+        CqlType::Map => CqlValue::Map(cql_value_vec_from_map(typ, &get_element!(Array))?),
+        CqlType::Set => CqlValue::Set(cql_value_vec_from_array(
+            typ.support_type_1
+                .as_ref()
+                .expect("Expected support type for list"),
+            &get_element!(Array),
+        )?),
+        CqlType::UserDefinedType => todo!(),
+        CqlType::SmallInt => CqlValue::SmallInt(
+            get_element!(i32)
+                .try_into()
+                .map_err(|_| js_error("Value must fit in i16 type to be small int"))?,
+        ),
+        CqlType::TinyInt => CqlValue::TinyInt(
+            get_element!(i32)
+                .try_into()
+                .map_err(|_| js_error("Value must fit in i8 type to be tiny int"))?,
+        ),
+        CqlType::Time => CqlValue::Time(get_element!(&LocalTimeWrapper).get_cql_time()),
+        CqlType::Timeuuid => CqlValue::Timeuuid(CqlTimeuuid::from_bytes(
+            get_element!(&UuidWrapper).get_cql_uuid().into_bytes(),
+        )),
+        CqlType::Tuple => CqlValue::Tuple(cql_value_vec_from_tuple(typ, &get_element!(Array))?),
+        CqlType::Uuid => CqlValue::Uuid(get_element!(&UuidWrapper).get_cql_uuid()),
+        CqlType::Varint => todo!(),
+        CqlType::Custom => return Err(js_error("Improper type information")),
+        CqlType::Empty => unreachable!("Should not receive Empty type here."),
+    };
+    Ok(value)
 }
 
-#[napi]
-impl MaybeUnsetQueryParameterWrapper {
-    #[napi]
-    pub fn from_non_null_non_unset_value(
-        val: &QueryParameterWrapper,
-    ) -> MaybeUnsetQueryParameterWrapper {
-        MaybeUnsetQueryParameterWrapper {
-            parameter: MaybeUnset::Set(val.parameter.clone()),
-        }
-    }
-
-    #[napi]
-    pub fn unset() -> MaybeUnsetQueryParameterWrapper {
-        MaybeUnsetQueryParameterWrapper {
-            parameter: MaybeUnset::Unset,
-        }
+impl FromNapiValue for ParameterWrapper {
+    /// # Safety
+    ///
+    /// Valid pointer to napi env must be provided
+    unsafe fn from_napi_value(
+        env: napi::sys::napi_env,
+        napi_val: napi::sys::napi_value,
+    ) -> napi::Result<Self> {
+        let parsing_error = || {
+            napi::Error::new(
+                Status::InvalidArg,
+                "Unexpected data when parsing parameters row".to_owned(),
+            )
+        };
+
+        // Caller of this function ensures a valid pointer to napi env is provided
+        let elem = unsafe { Array::from_napi_value(env, napi_val)? };
+        let val = match elem.len() {
+            0 => None,
+            1 => Some(MaybeUnset::Unset),
+            2 => {
+                let typ = elem.get::<&ComplexType>(0)?.ok_or_else(parsing_error)?;
+                Some(MaybeUnset::Set(cql_value_from_napi_value(typ, &elem, 1)?))
+            }
+            _ => {
+                return Err(parsing_error());
+            }
+        };
```

Comment:
:wrench: Please explain this logic in a comment (all the cases).

Reply:
I think you've already described it in internal docs, so let's just copy the relevant part here.